gnosis/test/verification.k

```diff
@@ -41,13 +43,14 @@ module VERIFICATION
 
     // select/store key equality
 
-    rule hash2(_,_) ==Int I => false
-      requires #isConcrete(I)
+    rule hash2(A,B) ==K I => false
+      requires         notBool #isConcrete(hash2(A,B)) andBool #isConcrete(I)
+               andBool I <Int 1000000000 //random small-ish number bigger than any fixed storage location
```

Comment:
i don't think you need `notBool #isConcrete(hash2(A,B))`, which is always true.

Reply:
Yeah' that's true. We actually need #isGround() - ensure that term doesn't have variables. Should we add a hook for it?

In the meantime, if all the rest is ok, can you merge please? This will help others get rid of non-determinism.

Reply:
i lost.  you mean that both A and B should NOT have a variable inside?  i don't think it should be the case.  any storage access with a function parameter value as an index will lead to the hash2 term with variables.

Reply:
Actually you are right. This side condition is not needed at all. Even if `#hash2(A,B)` is ground and should evaluate to a concrete number, we assume that number is very large, thus the last side condition ensures it cannot collide with I.

Deleted this side condition.